crates/api/src/func.rs

```diff
@@ -340,7 +349,7 @@ impl Func {
     ///     "#,
     /// )?;
     /// let instance = Instance::new(&module, &[add.into()])?;
-    /// let foo = instance.exports()[0].func().unwrap().get2::<i32, i32, i32>()?;
+    /// let foo = instance.exports().next().unwrap().into_func().unwrap().get2::<i32, i32, i32>()?;
```

Comment:
I've worried historicaly that this is pretty unwieldy. The previous incantation was sort of *just* on the line of not motivating me to try to improve things, but this feels like it's pretty unergonomic to document as a pattern to call things. Would you be ok adding more accessors in this PR for `Instance`? For example I think we could make this:

```rust
instance.get_func("foo")?.get2::<i32, i32, i32>()?;
```

Reply:
I also don't mind pushing up some new iterators/accessors for `Instance` as part of this PR too

Reply:
Good idea, I've now added `get_func` etc. accessors and update the examples and docs to use them, and that makes things much cleaner.